pkg/types/chain_reader.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"context"
+	"time"
+)
+
+// Errors exposed to product plugins
+
+type ErrChainReader string
+
+func (e ErrChainReader) Error() string { return string(e) }
+
+const (
+	ErrInvalidType   = ErrChainReader("invalid type")
+	ErrFieldNotFound = ErrChainReader("field not found")
+)
+
+// Errors used only by relay plugins
+
+type ErrorChainReaderUnsupported struct{}
```

Comment:
They are used in core, although I'm removing the usage rn

Reply:
Why is it important to share the same error? Is this better than wrapping the standard `ErrUnsupported`:
```go
return fmt.Errorf("ChainReader is not supported: %w", errors.ErrUnsupported)
```

Reply:
I'm fine with both tbh. I'll change it to `errors.ErrUnsupported`

Reply:
These are two very different types of errors, one is a user error where immediate action should be taken by the user. The other is a feature we don't support at all--I don't think we should conflate those two.

Reply:
Contemplating this more, I think I know what the source of your discomfort with all of this is:  we really ought to be sending these errors over the wire, otherwise there's no way for the UI to even report them.  I think it was a mistake to restrict them just to the relay and convert them to nil.  This should solve a number of issues @jmank88 has pointed out with this PR and the linked one.